orc8r/cloud/go/services/orchestrator/obsidian/models/conversion.go

```diff
@@ -55,6 +56,9 @@ func (m *Network) FromConfiguratorNetwork(n configurator.Network) *Network {
 	if cfg, exists := n.Configs[orc8r.NetworkSentryConfig]; exists && cfg != nil {
 		m.SentryConfig = cfg.(*NetworkSentryConfig)
 	}
+	if cfg, exists := n.Configs[orc8r.StateConfig]; exists && cfg != nil {
+		m.StateConfig = cfg.(*StateConfig)
+	}
```

Comment:
this check is a bit redundant, if StateConfig key is missing, cfg will be set at 'zero value' - nil anyway.
https://golang.org/doc/effective_go#maps
It's just a style preference, up to you...

Reply:
perhaps more interesting is the next line:

`m.StateConfig = cfg.(*StateConfig)`

should it be:

```
sc, ok := cfg.(*StateConfig)
if !ok {
  panic("wtf") // replace with useful message
}
m.StateConfig = sc
```

this would be along the lines of: "strict checks/patterns that strongly discourage the use of `map[string]interface{}`"

Reply:
makes sense - I followed the pattern the line above for NetworkSentry
```	
if cfg, exists := n.Configs[orc8r.NetworkSentryConfig]; exists && cfg != nil {
		m.SentryConfig = cfg.(*NetworkSentryConfig)
	}
```
but i agree we should be more defensive in how we approach casting.

Reply:
@emakeev removed the exists check
@markjen added a nil check to mappings in the  FromConfiguratorNetwork function.
I didn't add a panic since the code right now does not fail if only a partial mapping is made - introducing that can lead to unexpected outcomes